ops/model.py

```diff
@@ -839,33 +942,38 @@ class ConfigData(LazyMapping):
     This class should not be created directly. It should be accessed via :attr:`Model.config`.
     """
 
-    def __init__(self, backend):
+    def __init__(self, backend: '_ModelBackend'):
         self._backend = backend
 
     def _load(self):
         return self._backend.config_get()
 
 
-class StatusBase:
+class StatusBase(metaclass=ABCMeta):
     """Status values specific to applications and units.
 
     To access a status by name, see :meth:`StatusBase.from_name`, most use cases will just
     directly use the child class to indicate their status.
     """
 
-    _statuses = {}
-    name = None
+    _statuses = {}  # type: Dict[str, Type[StatusBase]]
 
-    def __init__(self, message: str):
+    def __init__(self, message: str = ''):
         self.message = message
 
-    def __new__(cls, *args, **kwargs):
+    @property
+    @abstractmethod
+    def name(self) -> str:
+        """Subclasses should override this attribute and make it a string."""
+        pass
```

Comment:
Rather than having the external error/raise check elsewhere, maybe we could just put a raise directly in here?

Reply:
resorted to making `name` NotImplemented and keeping the check in Register. It simplifies things